tests/pos/i8577/Main_2.scala

```diff
@@ -0,0 +1,164 @@
+package i8577
+
+def main: Unit = {
+  {
+    extension (ctx: StringContext) def macroA: MacroB.StringContext = MacroB(ctx)
+    extension (inline ctx: MacroB.StringContext) inline def unapplySeq(inline input: Int): Option[Seq[Int]] =
+      ${implUnapplyA('ctx, 'input)}
+
+    val macroA"$xA" = 1
+  }
+
+  {
+    extension (ctx: StringContext) def macroB: MacroB.StringContext = MacroB(ctx)
+    extension (inline ctx: MacroB.StringContext) inline def unapplySeq[U](inline input: U): Option[Seq[U]] =
+      ${ implUnapplyB('ctx, 'input) }
+
+    val macroB"$xB" = 2
+  }
+
+  {
+    extension (ctx: StringContext) def macroC: MacroC.StringContext = MacroC(ctx)
+    extension [T] (inline ctx: MacroC.StringContext) inline def unapplySeq(inline input: T): Option[Seq[T]] =
+      ${ implUnapplyC('ctx, 'input) }
+
+    val macroC"$xC" = 3
+  }
+
+  {
+    extension (ctx: StringContext) def macroD: MacroD.StringContext = MacroD(ctx)
+    extension [T] (inline ctx: MacroD.StringContext) inline def unapplySeq[U](inline input: T): Option[Seq[T]] =
+      ${ implUnapplyD('ctx, 'input) }
+
+    // miscompilation
+//    val macroD"$xD" = 4
```

Comment:
If the intentional that this should not compile?
If so, you could create a separate netagive test in `tests/neg/`